src/credentials/StaticGroupDataProvider.cpp

```diff
@@ -0,0 +1,514 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "GroupDataProvider.h"
+#include <lib/core/CHIPConfig.h>
+#include <string.h>
+
+using namespace chip::DeviceLayer;
+
+namespace chip {
+namespace Credentials {
+namespace {
+
+#ifndef CHIP_CONFIG_GROUP_DATA_FABRICS_MAX
+#define CHIP_CONFIG_GROUP_DATA_FABRICS_MAX 4
+#endif
+
+#ifndef CHIP_CONFIG_GROUP_DATA_MAPPINGS_MAX
+#define CHIP_CONFIG_GROUP_DATA_MAPPINGS_MAX 16
+#endif
+
+#ifndef CHIP_CONFIG_GROUP_DATA_STATES_MAX
+#define CHIP_CONFIG_GROUP_DATA_STATES_MAX 16
+#endif
+
+#ifndef CHIP_CONFIG_GROUP_DATA_KEY_SETS_MAX
+#define CHIP_CONFIG_GROUP_DATA_KEY_SETS_MAX 16
+#endif
+
+static constexpr size_t kGroupFabricsMax    = CHIP_CONFIG_GROUP_DATA_FABRICS_MAX;
+static constexpr size_t kEndpointEntriesMax = CHIP_CONFIG_GROUP_DATA_MAPPINGS_MAX;
+static constexpr size_t kStateEntriesMax    = CHIP_CONFIG_GROUP_DATA_STATES_MAX;
+static constexpr size_t kKeyEntriesMax      = CHIP_CONFIG_GROUP_DATA_KEY_SETS_MAX;
+
+class StaticGroupsProvider : public GroupDataProvider
+{
+    struct EndpointEntry : public GroupMapping
+    {
+        bool in_use = false;
+    };
+
+    struct StateEntry : public GroupState
+    {
+        bool in_use = false;
+    };
+
+    struct KeysEntry : public KeySet
+    {
+        bool in_use = false;
+    };
+
+    struct Fabric
```

Comment:
Suggest rename to `GroupContextForFabric` and `GetFabric` to `GetGroupContextForFabric`